### PR TITLE
Cli: Update OutputFormat method to return a String to restore consistency

### PR DESCRIPTION
**Pull request by CriesofCarrots**
#### Problem
The Cli `process_command()` method returns a String, which is printed in main.rs. In times past, every cli method populated this String, but as users wanted more complex and informative prints, they implemented those prints in the respective `process_` method and returned `Ok("".to_string())` to main.rs.

However, #9478 implemented Display for most of these complex prints, providing the opportunity to re-populate those empty Strings.

#### Summary of Changes
- Update OutputFormat to return a formatted String, instead of doing the print itself.
- Return formatted Strings from `process_command()`

Enables #9650 without the duplicated signature-prints

### CODE REVIEW

File: cli/src/cli.rs [hunk 1]
```diff
@@ -1265,7 +1265,7 @@ fn process_show_account(
         use_lamports_unit,
     };
 
-    config.output_format.formatted_print(&cli_account);
+    println!("{}", config.output_format.formatted_string(&cli_account));
```
Comment (t-nelson) [inline cli/src/cli.rs:1268]: We can use the same pattern here if we `formatted_string()` into a `mut String`, then change the `println!()`s below to `writeln!()`s.

Comment (CriesofCarrots) [reply cli/src/cli.rs:1268]: Ahhh, nice. I figured there was some way to fix this one, but didn't spend enough thoughts on it!

Comment (CriesofCarrots) [reply cli/src/cli.rs]: ✔️

**Comment by solana-grimes** [comment]
:broken_heart: Unable to automerge due to CI failure

**Comment by solana-grimes** [comment]
:broken_heart: Unable to automerge due to CI failure

**Pull request merged**